mongodbatlas/federation_settings.go

```diff
@@ -0,0 +1,109 @@
+// Copyright 2022 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mongodbatlas
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+)
+
+const federationSettingsBasePath = "api/atlas/v1.0/orgs/%s/federationSettings"
+const federationSettingsDeleteBasePath = "api/atlas/v1.0/federationSettings"
+
+// FederatedSettingsService is an interface for working with the Federation Settings
+// endpoints of the MongoDB Atlas API.
+// See more: https://www.mongodb.com/docs/atlas/reference/api/federation-configuration/
+type FederatedSettingsService interface {
+	Get(context.Context, string) (*FederatedSettings, *Response, error)
+	Delete(context.Context, string) (*Response, error)
+	ListConnectedOrgs(context.Context, *ListOptions, string) (*FederatedSettingsConnectedOrganizations, *Response, error)
+	GetConnectedOrgs(context.Context, string, string) (*FederatedSettingsConnectedOrganization, *Response, error)
+	UpdateConnectedOrgs(context.Context, string, string, *FederatedSettingsConnectedOrganization) (*FederatedSettingsConnectedOrganization, *Response, error)
+	DeleteConnectedOrgs(context.Context, string, string) (*Response, error)
+
+	ListRoleMapping(context.Context, *ListOptions, string, string) (*FederatedSettingsOrganizationRoleMappings, *Response, error)
+	GetRoleMapping(context.Context, string, string, string) (*FederatedSettingsOrganizationRoleMapping, *Response, error)
+	CreateRoleMapping(context.Context, string, string, *FederatedSettingsOrganizationRoleMapping) (*FederatedSettingsOrganizationRoleMapping, *Response, error)
+	UpdateRoleMapping(context.Context, string, string, string, *FederatedSettingsOrganizationRoleMapping) (*FederatedSettingsOrganizationRoleMapping, *Response, error)
+	DeleteRoleMapping(context.Context, string, string, string) (*Response, error)
+
+	ListIdentityProvider(context.Context, *ListOptions, string) ([]FederatedSettingsIdentityProvider, *Response, error)
```

Comment:
```suggestion
	ListIdentityProviders(context.Context, *ListOptions, string) ([]FederatedSettingsIdentityProvider, *Response, error)
```

Reply:
Refactored to ListIdentityProviders(context.Context, string, *ListOptions) ([]FederatedSettingsIdentityProvider, *Response, error)